src/physics/navier_stokes.cpp

```diff
@@ -116,6 +116,47 @@ ::compute_temperature_gradient (
     return temperature_gradient;
 }
 
+template <int dim, int nstate, typename real>
+template<typename real2>
+real2 NavierStokes<dim,nstate,real>
+::compute_wall_shear_stress (
+    const std::array<real2,nstate> &conservative_soln,
+    const std::array<dealii::Tensor<1,dim,real2>,nstate> &conservative_soln_gradient,
+    const dealii::Tensor<1,dim,real2> &normal_vector) const
+{
+    // Computes the non-dimensional wall shear stress
+    // NOTE: Currently this is only implemented for channel flow
+
+    // get primitive solution
+    const std::array<real2,nstate> primitive_soln = this->template convert_conservative_to_primitive<real2>(conservative_soln); // from Euler
+    // extract from primitive solution
+    // const dealii::Tensor<1,dim,real2> velocities = this->template extract_velocities_from_primitive<real2>(primitive_soln); // from Euler
+    const std::array<dealii::Tensor<1,dim,real2>,nstate> primitive_soln_gradient
+                 = this->template convert_conservative_gradient_to_primitive_gradient<real2>(conservative_soln,conservative_soln_gradient);
+    const dealii::Tensor<2,dim,real2> velocities_gradient = extract_velocities_gradient_from_primitive_solution_gradient<real2>(primitive_soln_gradient);
+
+    // // compute normal velocity
+    // real2 wall_parallel_velocity = 0.0;
+    // for(int d=0;d<dim;++d){
+    //     normal_velocity += velocities[d]*normal_vector[d];
+    // }
+    // // compute wall parallel velocity
+    // dealii::Tensor<1,dim,real2> velocities_parallel_to_wall;
+    // for(int d=0;d<dim;++d){
+    //     velocities_parallel_to_wall[d] = velocities[d] - normal_velocity*normal_vector[d];
+    // }
+
+    const real2 scaled_viscosity_coefficient = compute_scaled_viscosity_coefficient<real2>(primitive_soln);
+    real2 velocity_gradient_of_parallel_velocity_in_the_direction_normal_to_wall = 0.0;
+    for(int d=0;d<dim;++d){
+        velocity_gradient_of_parallel_velocity_in_the_direction_normal_to_wall += velocities_gradient[0][d]*normal_vector[d];
+    }
+    // Reference: https://www.cfd-online.com/Wiki/Wall_shear_stress
+    const real2 wall_shear_stress = scaled_viscosity_coefficient*velocity_gradient_of_parallel_velocity_in_the_direction_normal_to_wall;
```

Comment:
Is this computed in the computational or physical domain?

Reply:
physical